### ISSUE TITLE
Add ability to sort/reorder loadouts

### ISSUE THREAD

**Issue by nguyentvan7** [frzyc/genshin-optimizer#1725]
There probably isn't a great way to do this automatically using heuristic, so maybe just allow custom sorting, similar to multi opt?

**Comment by HassnHamada** [comment]
I'll work on this

**Comment by frzyc** [comment]
Yeah will probably need more thought here, because there isnt an actual "array" to persist, so this is more complicated than other sort/reorder, and there isnt a meaningful "last edit" either. I'd say we can defer this until a more evident use case arises.

**Comment by HassnHamada** [comment]
I thought about it and dnd sound like the better idea. So I'll just take leave it to him.

---
Just to share thoughts these are the two ways I would have done it

## To Manually Order
The way I would have done it is using arrows (MoveUp/MoveDown/MoveToStart/MoveToEnd).

Which got two problems
- It doesn't help if someone wants to move only half way (They would have to click multiple times)
- I don't believe it will look good and will make the UI more crowded (Will have to add 4 buttons for 1 functionality)

## To Automatically Sort
One solution is to displaying the value for the target being currently optimized next to the build name and sort base on that. It's only problem people may not like it since It may move a certain build from a certain location they are used to, essentially confusing them.

**Comment by frzyc** [comment]
Yeah, I don't think a "custom sort order" is the way to go here, just a lot of extra UI and complicated logic. The best we can do is sort by loadout name (non-persist and dynamic), but I am honestly not sure what people expect here in terms of sort, hence why I want more feedback before doing any engineering time.
